### PR TITLE
support ens/kns domains for pledges

### PR DESCRIPTION
**Pull request by 0xAeterno**
## Description

Add kns/ens support to pledge dashboard pages.
Pledge profile pictures will prioritise image sources in the following order:
- Pledge
- KNS custom avatar
- KNS default/fallback avatar
- ENS

TODO:
- [x] Resolve kns/ens profile pictures
- [x] Search domains from `/pledge`?

Minor fixes:
- Hide placeholder text on input focus (/pledge)
- Met pledges will now display `"≥ 100% of pledge met"` 
- Text casing

## Related Ticket

#489 

Test pledges
- [KNS - /pledge/address](https://klimadao-site-git-pledge-ens-kns-support-klimadao.vercel.app/pledge/0x293Ed38530005620e4B28600f196a97E1125dAAc) - address with kns
- [KNS - /pledge/kns-domain](https://klimadao-site-git-pledge-ens-kns-support-klimadao.vercel.app/pledge/archimedes.klima) - kns without pledge but kns avatar is set
- [KNS - /pledge/kns-domain](https://klimadao-site-git-pledge-ens-kns-support-klimadao.vercel.app/pledge/markcuban.klima) - kns without pledge or kns avatar 
- [ENS - /pledge/ens-domain](https://klimadao-site-git-pledge-ens-kns-support-klimadao.vercel.app/pledge/matoken.eth) - with avatar set

Note: ENS avatar resolution takes a long time :/

## Checklist

- [x] Building the site with `npm run build-site` works without errors
- [x] Building the app with `npm run build-app` works without errors
- [x] I formatted JS and TS files with running `npm run format-all`

### CODE REVIEW

File: site/pages/pledge/[address].tsx [hunk 1]
```diff
@@ -10,8 +12,16 @@ export const getStaticProps: GetStaticProps = async (ctx) => {
...
   let pledge;
+  let resolvedAddress;
+
+  const isDomainInURL = getIsDomainInURL(address);
+  const domain = isDomainInURL ? address : null;
 
-  if (!ethers.utils.isAddress(address)) {
+  try {
+    resolvedAddress = isDomainInURL
+      ? await getAddressByDomain(address)
```
Comment (0xAeterno) [inline site/pages/pledge/[address].tsx:22]: @Atmosfearful
This has got me stumped, exceptions from a non ens/kns domain in the url aren't being caught and redirected as expected. Any ideas on this one?

image

Comment (0xAeterno) [reply site/pages/pledge/[address].tsx:22]: Ah I think I worked it out. I just needed to set an Infura ID ☺️

Comment (ladytrekker) [reply site/pages/pledge/[address].tsx:24]: Yes exactly! 
This error should only appear when you are trying to resolve an ENS address without the INFURA ID.

But you said above that this error happened also on non ENS addresses? Or did I got that wrong?

File: site/pages/pledge/[address].tsx [hunk 2]
```diff
@@ -21,21 +36,20 @@ export const getStaticProps: GetStaticProps = async (ctx) => {
...
   }
 
   return {
     props: {
-      pageAddress: address.toLowerCase(),
-      pledge: {
-        ...pledge,
-        ownerAddress: pledge?.ownerAddress || address.toLowerCase(),
-      },
+      canonicalUrl: `${urls.pledges}/${address}`,
```
Comment (Atmosfearful) [inline site/pages/pledge/[address].tsx:49]: I think I may have gave some bad advice a few weeks ago when thinking about the retirement urls

I originally thought we wanted to point search engines from `/pledge/0x123` to the .klima or .ens version, if it exists
But I don't think this is a good idea anymore, so we just do self-referential urls as you did here. Not really doing much afaik.

We should revisit at some point

File: site/components/pages/Pledge/index.tsx [hunk 3]
```diff
@@ -44,15 +44,15 @@ export const Pledge: NextPage = () => {
         <form className={styles.inputContainer} onSubmit={handleFormSubmit}>
           <input
             className={styles.input}
-            placeholder="Search for a wallet address"
+            placeholder="Enter an address or an ENS/KNS domain"
```
Comment (Atmosfearful) [inline site/components/pages/Pledge/index.tsx:47]: Would this be more intuitive for non-defi normies?
```suggestion
            placeholder="Enter a wallet address, .eth or .klima domain"
```

**Review by Atmosfearful — APPROVED** [review]

**Pull request merged**